app/controllers/npq_separation/admin/statements_controller.rb

```diff
@@ -0,0 +1,84 @@
+class NpqSeparation::Admin::StatementsController < NpqSeparation::AdminController
+  before_action :ensure_super_admin
+  before_action :set_cohort
+  before_action :store_uploads, only: :create
+
+  def new; end
+
+  def create
+    @service = Statements::BulkCreator.new(
+      cohort: @cohort,
+      statements_csv_id: bulk_creator_params[:statements_csv_id],
+      contracts_csv_id: bulk_creator_params[:contracts_csv_id],
+    )
+
+    @statements = @service.call(dry_run:)
+
+    if @service.errors.any?
+      render :new
+    elsif dry_run
+      set_preview
+      render
+    else
+      flash[:success] = "#{@statements.count} statements created successfully"
+      redirect_to npq_separation_admin_cohort_path(@cohort)
+    end
+  end
+
+  def show
+    row_class = {
+      "statements" => Statements::BulkCreator::StatementRow,
+      "contracts" => Statements::BulkCreator::ContractRow,
+    }.fetch(params[:id])
+
+    render plain: row_class.example_csv.lines.first, content_type: "text/csv", filename: "#{params[:id]}.csv"
```

Comment:
I'd expect csv would already be registered in mime types (so `content_type: :csv`) but if its not lets add it there

Reply:
Thinking on this shouldn't this (and other similar lines) be `send_data` ?